### DISCUSSION TITLE
Turning with Prime hub and Xbox controller

### DISCUSSION THREAD

**Discussion by sofiania** [pybricks/support#1631] [Q&A]
I have two Spike Prime cars, eaxch with two motors driving (A and B) and one motor steering (F). The first car is remote controlled with another Prime hub and the other with the Xbox, but the one controlled with the Prime hub makes better turns. With the xbox, it drifts mor and makes larger turns. Does anyone have ideas on how to refine the xbox code to make tighter turns?
I use the code below (ignore the "force" variable).
![image](https://github.com/pybricks/support/assets/35339701/f2c6a5da-e2a3-4687-809d-b641d28dfef2)
![image (3)](https://github.com/pybricks/support/assets/35339701/b4d7f77e-7cad-4f0a-b2f2-8150b0cea3d8)

**Comment by laurensvalk** [comment]
The "left joystick horizontal" produces a value from -100 to 100. That's the same as needed for steering.

But if you'd like it to be more responsive, so steer 100% even if you push the joystick only halfway, say 50%, you could multiply the value by `2`, very similar to how you did it in the other program. For example:

![image](https://github.com/pybricks/support/assets/12326241/765e181e-ee34-4d7d-9fa0-62877ea12980)

But it will still only turn 100% at the most, limited by the mechanical design. Are both cars the same?

**Reply by sofiania** [reply]
That didn't make much of a change.... I guess i'll just have to drive a bit slower :). Yes the cars are exactly the same. It's the hot rod from Anton's mindstorm. Aaand ha has a very complicated code for the lego Spike app with torque, snd DC motor power and all to go with it, but I don't have enough time to understand it enough to do the same in pybricks (plus I work with kids, so wanted a bit less complex code :) . 
![Skärmbild 2024-05-08 095531](https://github.com/pybricks/support/assets/35339701/0b8979dd-1b1c-4d3e-9697-c4cbb44bd5b6)

**Reply by laurensvalk** [reply]
I think it might have to do with the few degrees of margin we take at 100% in the simplified car block, so we don't push the steering motor too much. For a car with the motor linked directly to the steering, I think we can relax that margin a bit. This is something we can do in a future update :)

Edit: suggested solution below.

**Reply by laurensvalk** [reply]
Actually, you could do it even simpler like this! Keep your original program and replace only the steering block. This gives you full control over the maximum steering angle. See the comment below:

![image](https://github.com/pybricks/support/assets/12326241/33bfa5f6-9039-4b52-af16-90638ee0d39d)

**Reply by sofiania** [reply]
Nope, unfortunately not near as good turns as the hub controlled with another hub :( unless there actually is something wrong with the second car that is...like with the motors. I'll try to connect the first car to xbox instead to rule that out.

**Reply by sofiania** [reply]
No, it was the same on the first car... But it works if I go more slowly of course.

**Reply by sofiania** [reply]
I just realised that there was a Pybricks block program in the zipfile from the package I bought from Anton's mindstorm page, and that code solved the issue (but made the car drive a bit slower). I don't think I'm allowed to post that code here though, since I paid for acess to the content.
